spring-integration-file/src/test/java/org/springframework/integration/file/tail/FileTailingMessageProducerTests.java

```diff
@@ -124,13 +131,48 @@ public void canRecalculateCommandWhenFileOrOptionsChanged() throws IOException {
 		adapter.stop();
 	}
 
+	@Test
+	@TailAvailable
+	public void testIdleEvent() throws Exception {
+		OSDelegatingFileTailingMessageProducer adapter = new OSDelegatingFileTailingMessageProducer();
+		adapter.setOptions(TAIL_OPTIONS_FOLLOW_NAME_ALL_LINES);
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.afterPropertiesSet();
+		adapter.setTaskScheduler(taskScheduler);
+		CountDownLatch countDownLatch = new CountDownLatch(1);
+		adapter.setApplicationEventPublisher(event -> {
+			if (event instanceof FileTailingIdleEvent) {
+				FileTailingIdleEvent tailEvent = (FileTailingIdleEvent) event;
+				logger.debug(event);
+				countDownLatch.countDown();
+			}
+		});
+		File file = new File(testDir, "foo");
+		file.delete();
+		adapter.setFile(file);
+		QueueChannel outputChannel = new QueueChannel();
+		adapter.setOutputChannel(outputChannel);
+		adapter.setIdleEventInterval(100);
+		adapter.afterPropertiesSet();
+		adapter.start();
+		boolean noEvent = countDownLatch.await(500, TimeUnit.MILLISECONDS);
```

Comment:
i am not sure we can skip this wait, because even if we catch the no file exist event without any other event that doesn't mean `FileTailingIdleEvent` is not going to be emitted later on , we have to wait till `IdleEventInterval`  pass and then check no event is emitted .

Unless i am missing some thing , i think we have to wait at least the `IdleEventInterval`